### ISSUE TITLE
[Feature Request] EarlyStopping callback patience for stopping_threshold

### ISSUE THREAD

**Issue by yelwah** [open-mmlab/mmdetection#11575]
**Feature**
The `EarlyStopping` `Callback` module currently has a feature that if the `monitor` metric does not improve over for at least `patience` epochs, then stop training. I purpose a similar feature for the `stopping_threshold`, that if the metric stays above/below the threshold for at least `threshold_patience` epochs, then stop training.  

**Motivation**
Certain metrics either do not converge, or may start drifting as overfitting begins to occur. Additionally, that metric may hit the target value briefly earlier in training, at which point the model is still underfit. By determining that the metric has been achieved, but only for a certain number of epochs, it allows for fast training, while still staying beyond a target threshold signifying model accuracy.

This has all been stated in the affirmative case, as a metric you want to be above/below a threshold, but could also be used for the negative case of stopping if a certain metric is beyond a certain threshold that may signify the model is far off. Additionally, like the affirmative case, a certain number of epochs beyond the threshold may be permissible, but if its beyond that threshold for some length of time (`threshold_patience`) it's not longer viable and training should be stopped.
